### DISCUSSION TITLE
Dynamic relation

### DISCUSSION THREAD

**Discussion by Gsom1** [symfony/symfony#47685] [Q&A] [answered]
Is it possible to have ManyToOne relation, where one is distributed across many tables depending on type.

Let's the structure is like:
```
        $this->addSql('CREATE TABLE market_item (
            id serial,
            item_id UUID NOT NULL,
            item_type VARCHAR(255) NOT NULL
        PRIMARY KEY(id))'
        );
        
       $this->addSql('CREATE TABLE rooster (
            id UUID NOT NULL,
            user_id INT NOT NULL,
        PRIMARY KEY(id))
        ');

       $this->addSql('CREATE TABLE chicken(
            id UUID NOT NULL,
            user_id INT NOT NULL,
        PRIMARY KEY(id))
        ');
```

so an item can relate to either chicken or rooster

**Comment by MatTheCat** [comment, marked as answer]
See https://www.doctrine-project.org/projects/doctrine-orm/en/2.13/reference/inheritance-mapping.html

If your chicken and rooster entities inherit from a same class you can use it as the target of the association. Be aware this has performance impacts though.

**Reply by Gsom1** [reply]
yep @Discriminator% looks very relevant I'll try it
Thank you so much!

**Reply by Gsom1** [reply]
so I tried to use InheritanceType('TABLE_PER_CLASS') and got `This behaviour is (currently) not supported by Doctrine 2`
apparently I need to review me data structure.

**Reply by MatTheCat** [reply]
Where did you find `TABLE_PER_CLASS`? The documentation I linked shows `JOINED`.

**Reply by Gsom1** [reply]
There is a class Doctrine\ORM\Mapping\ClassMetadataInfo TABLE_PER_CLASS is inside.
Generally inheritance assumes market_item->id has a FK to (rooster|chicken)->id which is not like this in my case.
May be my design is wrong I'm not sure yet.

Let me bring a little bit context here.
I'm implementing a shop, market_item table contains items being sell, so these can be absolutely different types,
so I guess the inheritance is not the best choice here.

an sql query to fetch items would be like this:

```
select mi.*, COALESCE (c.user_id, r.user_id, …)  user_id from market_items mi
left join chicken c on c.id = mi.item_id and mi.item_type = 'chicken'
left join rooster r on c.id = mi.item_id and mi.item_type = 'rooster'
```

**Reply by MatTheCat** [reply]
> There is a class Doctrine\ORM\Mapping\ClassMetadataInfo TABLE_PER_CLASS is inside.

You should consider the documentation instead of your personal understanding of the code.

> I'm implementing a shop, market_item table contains items being sell, so these can be absolutely different types

All these types are items being sell so here is your base type.
